portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -0,0 +1,34 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/")
+public class HomeServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html");
+
+        UserService userService = UserServiceFactory.getUserService();
+
+        if (userService.isUserLoggedIn()) {
+            String userEmail = userService.getCurrentUser().getEmail();
+            String urlToRedirectToAfterUserLogsOut = "/";
+            String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
```

Comment:
...although to immediately add to my previous comment, you may not need a variable for the post logout redirect URL if you are only using it one place, and that place is the very next line.

Also, considering that you have another variable for the URL to go to after logging *in*, and that URL is the exact same, maybe consider just consider creating a class constant:
private static final String ROOT_URL = "/";

and just using that same constant in both places, since it seems like in both cases you want to go back to the root/home page.

Reply:
Resolved!